rfd/0001-tooling-requirements.md

```diff
@@ -0,0 +1,151 @@
+
+Projects should have the following resources in their directory:
+* A `README.md` document at the project root. This should detail what the project is and how to run it, including a least one explicit example. This should also contain information on any major nuances that end users should be aware of. For example, a README for the OS package repo tool would document that the tool should never be ran concurrently when targeting the same S3 bucket.
+* A `CHANGELOG.md` document that follows the standard [described here](https://keepachangelog.com/).
+* A `renovate.json5` file that configures any [Renovate managers](https://docs.renovatebot.com/modules/manager/) required to keep the project's dependencies up to date.
+* An `action.yaml` file (optional) for tools that support being ran as Github actions. This action should _not_ compile the tool, download dependencies, or perform any other actions that could reasonably be completed once per release. In the case of non-composite actions, this is intended to be a simple GHA-friendly wrapper of the underlying tool.
+* A `.gitignore` file (optional) that contains project-specific filepath patterns that Git should ignore.
+* A `docs` directory (optional) which includes additional useful information regarding the project that is not included in the README.
+* A `workflows` directory which all Github workflows related to the project's lifecycle. This includes (at minimum) a CI workflow for validating PR changes, a CD workflow for releasing dev tag versions upon PR merge, and a release/promotion workflow for promoting a dev tag release when a new version is added to the changelog. 
+
+#### CI workflow
+The status check workflows should not trigger upon changes to specific paths, rather, they should follow the precedent set in the `gravitational/cloud` repo and run on every PR that is opened. Each workflow should then use an action [like this one](https://github.com/dorny/paths-filter) to determine whether or not the check should run, or be bypassed. This removes the need for the superfluous "bypass" workflow pattern used in other repositories in the Gravitational org.
+
+#### CD workflow
+This workflow should run when a PR is merged, and optionally upon PR commits as a way to produce test builds. When ran, this workflow should upload artifacts as Github Packages where possible. These artifacts should not be uploaded to AWS unless absolutely necessary, and with Security's permission, as outlined in the [Security considerations section](#security-considerations). The name of these artifacts should include a dev tag generated by `git describe --tags` to ensure uniqueness.
+
+#### Release workflow
+The release workflow should be triggered when a new version is added to the CHANGELOG on `main` branch. When this occurs, the workflow should:
+1. Extract the new version from the CHANGELOG.
```

Comment:
Seems a little strange to trigger release on changelog update, why not trigger it on a tag?

Reply:
I don't think this is strictly needed. Updated to trigger on tag.